f3/granite.go

```diff
@@ -266,28 +276,32 @@ func (i *instance) receiveOne(msg *GMessage) {
 	if msg.Step == COMMIT && i.phase != DECIDE {
 		i.tryCommit(msg.Round)
 	} else {
-		i.tryCompletePhase()
+		return i.tryCompletePhase()
 	}
+
+	return nil
 }
 
 // Attempts to complete the current phase and round.
-func (i *instance) tryCompletePhase() {
+func (i *instance) tryCompletePhase() error {
 	i.log("try step %s", i.phase)
 	switch i.phase {
 	case QUALITY:
-		i.tryQuality()
+		return i.tryQuality()
 	case CONVERGE:
-		i.tryConverge()
+		return i.tryConverge()
 	case PREPARE:
-		i.tryPrepare()
+		return i.tryPrepare()
 	case COMMIT:
 		i.tryCommit(i.round)
+		return nil
```

Comment:
Make both tryCommit and tryDecide return `error` too, even if always nil. We'll probably need it later and can avoid this noise when adding some specific error.

Reply:
Yes I was actually thinking about exactly that, wondering which option is better. This is exactly the nudge that was missing to actually do it :-)